data/functions/itemized.sql

```diff
@@ -16,3 +124,22 @@ begin
     perform retry_processing_schedule_b_records(:START_YEAR_AGGREGATE);
 end
 $$ language plpgsql;
+
+CREATE OR REPLACE FUNCTION rename_table_cascade(table_name TEXT) RETURNS VOID AS $$
+DECLARE
+    child_tmp_table_name TEXT;
+    child_tables_cursor CURSOR (parent TEXT) FOR
+        SELECT c.oid::pg_catalog.regclass AS name
+        FROM pg_catalog.pg_class c, pg_catalog.pg_inherits i
+        WHERE c.oid=i.inhrelid
+            AND i.inhparent = (SELECT oid from pg_catalog.pg_class rc where relname = parent);
+BEGIN
+    EXECUTE format('DROP TABLE IF EXISTS %I CASCADE', table_name);
+    EXECUTE format('ALTER TABLE %1$I_tmp RENAME TO %1$I', table_name);
+
+    FOR child_tmp_table IN child_tables_cursor(table_name) LOOP
+        child_tmp_table_name = child_tmp_table.name;
+        EXECUTE format('ALTER TABLE %1$I RENAME TO %2$I', child_tmp_table_name, regexp_replace(child_tmp_table_name, '_tmp$', ''));
+    END LOOP;
+END
+$$ LANGUAGE plpgsql;
```

Comment:
I think I had originally envisioned this as being a simpler implementation - we more or less just hard code the master table and child table names in the creation of the partition and just substitute the "a" and "b" portions in as needed, and I saw the renaming of them at the end working in a similar way in my mind's eye.

That said, I see that this can be used for any partition setup that requires the renaming to support a zero-down-time update, which is awesome! :-)